### ISSUE TITLE
[Bug]: Having some problems with pip installation: as of now->ImportError: cannot import name 'to_dtype' from 'torchvision.transforms.functional'

### ISSUE THREAD

**Issue by federicocaccialanzaabb** [openvinotoolkit/anomalib#1930]
### Describe the bug

I have installed anomalib version 1.0.0 with the command
pip install anomalib==1.0.0

and when I tried to import:

from anomalib.data.utils import read_image
from anomalib.deploy import OpenVINOInferencer

from notebooks use cases part a (but I tried as well the getting started one)
I run in several import libraries issues such as missing typing_extensions or lightning or imgaug or others

so I installed the core_requirements.txt that I found in the repo but I am still having some issues, currently:
 ImportError: cannot import name 'to_dtype' from 'torchvision.transforms.functional'

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

pip install -r core.txt
pip install anomalib==1.0.0

from anomalib.data.utils import read_image

ImportError: cannot import name 'to_dtype' from 'torchvision.transforms.v2.functional'

### OS information

OS information:
- Anomalib version: 1.0.0
- pip version: 22.3.1

### Expected behavior

I would like to be able to import the library

### Pip/GitHub

pip

### What version/branch did you use?

anomalib 1.0.0

### Configuration YAML

```yaml
I am not currently using any configuration YAML
```

### Logs

```shell
ImportError: cannot import name 'to_dtype' from 'torchvision.transforms.v2.functional' 
File <command-2245495519066745>, line 4
      1 """501a_training_a_model_with_cubes_from_a_robotic_arm.ipynb."""
      3 from pathlib import Path
----> 4 from anomalib.data.utils import read_image
      5 from anomalib.deploy import OpenVINOInferencer
File /local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/anomalib/data/utils/image.py:20
     18 from torch.nn import functional as F  # noqa: N812
     19 from torchvision.datasets.folder import IMG_EXTENSIONS
---> 20 from torchvision.transforms.v2.functional import to_dtype, to_image
     21 from torchvision.tv_tensors import Mask
     23 from anomalib.data.utils.path import validate_path
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by samet-akcay** [comment]
@federicocaccialanzaabb, we have just uploaded v1.0.1. Can you try that one by running the following?

```bash
pip install anomalib
anomalib install 
```

Also can you show me the output of the following?

```bash
anomalib --help
```

**Comment by federicocaccialanzaabb** [comment]
Hi @samet-akcay thank you for your fast reply.
I installed the version 1.0.1 (and also run anomalib install) but the issue unfortunately still persists:
`ImportError: cannot import name 'to_dtype' from 'torchvision.transforms.v2.functional' `
Another information that may be useful the Python version is 3.10.12

here is the output of anomalib --help 
ps: I can see that there is one of the first issue that I had which was the typing_extensions import erro
```
Traceback (most recent call last):
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/bin/anomalib", line 5, in <module>
    from anomalib.cli.cli import main
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/anomalib/cli/__init__.py", line 6, in <module>
    from .cli import AnomalibCLI
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/anomalib/cli/cli.py", line 14, in <module>
    from jsonargparse import ActionConfigFile, ArgumentParser, Namespace
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/__init__.py", line 12, in <module>
    from ._cli import *  # noqa: F403
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_cli.py", line 7, in <module>
    from ._core import ArgumentParser
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_core.py", line 48, in <module>
    from ._formatters import DefaultHelpFormatter, empty_help, get_env_var
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_formatters.py", line 27, in <module>
    from ._link_arguments import ActionLink
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_link_arguments.py", line 22, in <module>
    from ._parameter_resolvers import get_signature_parameters
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_parameter_resolvers.py", line 18, in <module>
    from ._stubs_resolver import get_stub_types
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_stubs_resolver.py", line 16, in <module>
    tc = import_typeshed_client()
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/jsonargparse/_optionals.py", line 71, in import_typeshed_client
    import typeshed_client
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/typeshed_client/__init__.py", line 4, in <module>
    from .finder import (
  File "/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/typeshed_client/finder.py", line 25, in <module>
    from typing_extensions import deprecated
ImportError: cannot import name 'deprecated' from 'typing_extensions' (/local_disk0/.ephemeral_nfs/envs/pythonEnv-df4f6f3b-65d9-4fd1-89b8-2f4088124389/lib/python3.10/site-packages/typing_extensions.py)
```

**Comment by samet-akcay** [comment]
can you show your torchvision version?

**Comment by federicocaccialanzaabb** [comment]
yes, of course,

running pip list 

torch                          2.1.2+cpu
torchmetrics                   1.3.2
torchvision                    0.16.2+cpu

**Comment by samet-akcay** [comment]
@federicocaccialanzaabb, there is a similar issue here
https://github.com/tiangolo/fastapi/discussions/9808

Which virtualenv tool are you using?

Also, people seem to fix it by re-installing `typing_extensions` on their environment

**Comment by federicocaccialanzaabb** [comment]
Thanks @samet-akcay, resetting the venv and reinstalling everything with forcing the correct version of typing_extensions worked,

Thanks again for the fast support on this issue

**Comment by skaderbe** [comment]
We faced a similar issue during installation. Resolved by following steps listed below.

Create a virtual environment in Anaconda root command prompt,
conda create -n anomalib_env
In Anaconda navigator, goto anomalib_env and install command prompt and spyder
Launch anomalib_env command prompt and follow the following sequence of steps in anomalib_env command prompt.
pip install anomalib
anomalib install
pip install lightning
pip install torchvision
pip install opencv-python
pip install imgaug
pip install kornia
pip install pandas
pip install scikit-learn
pip install timm
pip install einops
pip install FrEIA
pip install open_clip_torch
pip install OpenVINO

Simultaneously launch anomalib_env spyder and execute the code listed in "Training" section of Anomalib webpage.
